### PR TITLE
Add Runtime_Check and Static_Check interfaces

### LINKED ISSUE #60: Create Runtime_Check and Static_Check interfaces
## Description
Create 2 new interfaces:
- `Runtime_Check` interface.
- `Static_Check` interface.
 
These interfaces are used by Runner classes to determine the type of Check is being run.

## Acceptance Criteria
- A `Runtime_Check` interface class exists in `includes/Checker/Runtime_Check.php`.
- The `Runtime_Check` extends the `Check` interface.
- The `Runtime_Check` interface has no methods.
- The `Abstract_Runtime_Check` class implements the `Runtime_Check` interface.
- A `Static_Check` interface class exists in `includes/Checker/Static_Check.php`.
- The `Static_Check` extends the `Check` interface.
- The `Static_Check` interface has no methods.
-  The `Abstract_PHP_CodeSniffer_Check` class implements the `Static_Check` interface.

Additionally update the following classes:
- `Abstract_Runtime_Check` implements the `Runtime_Check` interface (remove the `Check` interface)
- `Abstract_PHP_CodeSniffer_Check ` implements the `Static_Check` interface (remove the `Check` interface)

### PR DESCRIPTION
**Pull request by vishalkakadiya**
Adds the Runtime_Check and Static_Check interfaces

- Addresses the acceptance criteria in https://github.com/10up/plugin-check/issues/60

Closes https://github.com/10up/plugin-check/issues/60

### CODE REVIEW

File: includes/Checker/Runtime_Check.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+/**
+ * Interface WordPress\Plugin_Check\Checker\Runtime_Check
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker;
+
+/**
+ * Interface for Runtime checks.
```
Comment (felixarntz) [inline includes/Checker/Runtime_Check.php]: Nit-pick:
```suggestion
 * Interface for runtime checks.
```

File: includes/Checker/Static_Check.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+/**
+ * Interface WordPress\Plugin_Check\Checker\Static_Check
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker;
+
+/**
+ * Interface for Static checks.
```
Comment (felixarntz) [inline includes/Checker/Static_Check.php]: Nit-pick:
```suggestion
 * Interface for static checks.
```

**Review by felixarntz — APPROVED** [review]
@vishalkakadiya This is pretty much good to go, so I already approved it. Just a small nit-pick that would be nice to address (unnecessary capitalization).

File: includes/Checker/Static_Check.php (hunk 2 above)
Comment (vishalkakadiya) [reply includes/Checker/Static_Check.php]: Updated this now. Thank you!

File: includes/Checker/Runtime_Check.php (hunk 1 above)
Comment (vishalkakadiya) [reply includes/Checker/Runtime_Check.php]: Updated this now. Thank you!

**Comment by felixarntz** [comment]
@jjgrainger When you have a moment, can you review this PR please?

**Review by jjgrainger — APPROVED** [review]

**Comment by jjgrainger** [comment]
Approved and merged, thanks @vishalkakadiya

**Comment by vishalkakadiya** [comment]
Thank you so much! @jjgrainger @felixarntz

**Pull request merged**